packages/material-ui-lab/src/useAutocomplete/useAutocomplete.js

```diff
@@ -607,8 +607,7 @@ export default function useAutocomplete(props) {
     selectNewValue(event, filteredOptions[highlightedIndexRef.current]);
   };
 
-  const handleTagDelete = event => {
-    const index = Number(event.currentTarget.getAttribute('data-tag-index'));
+  const handleTagDelete = index => event => {
     const newValue = [...value];
     newValue.splice(index, 1);
```

Comment:
@oliviertassinari Just wanted to know, why are we using splice here. For example i am having 3 values selected [1,2,3] and trying to delete first element in the array, in this case we will be passing index as 0 to renderTagProps. So if you splice array with 0,1 (newValue.splice(0,1)) will return first element in the array [1].  

We are facing issue like if we try to delete any element in our selected value list, it always removes last element instead of actual element which we try to delete. 

![image](https://github.com/mui/material-ui/assets/13847501/b08ce487-14aa-44fc-b138-50cffbaaddba)